### PR TITLE
Add validation to attributeFilter operations

### PR DESCRIPTION
**Pull request by wcmitchell**
This is about the most basic way to do this. I considered adding additional attributeFilter validation, but it seems the rest is working as is, and if more needs show up later we can build on this framework.

Signed-off-by: Chris Mitchell <cmitchel@redhat.com>

## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-5651

## Description of Intent of Change(s)
APIspec generated clients need us to validate the 'operation' value in provided ResourceDefinition attributeFilters

## Local Testing

1. Pull down branch
2. make serve
3. Create a role with a valid resourceDefinition attributeFilter operation: 
` curl localhost:8000/api/rbac/v1/roles/ -X POST -H 'content-Type: application/json' -d '{"name": "testy", "access": [{"permission": "cost-management:*:read", "resourceDefinitions": [{"attributeFilter":{"key":"sa","value":"sda","operation":"in"}} ]}]}'`
4. Verify above worked
5. Create a role with an invalid operation:
 curl localhost:8000/api/rbac/v1/roles/ -X POST -H 'content-Type: application/json' -d '{"name": "testy2", "access": [{"permission": "cost-management:*:read", "resourceDefinitions": [{"attributeFilter":{"key":"sa","value":"sda","operation":"sad"}} ]}]}'
6. Verify failure of above creation

## Checklist
- [ ] if API spec changes are required, is the spec updated?

- [ ] are there any pre/post merge actions required? if so, document here.

- [x] are theses changes covered by unit tests?

- [ ] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

File: tests/management/role/test_view.py [hunk 1]
```diff
@@ -212,6 +212,30 @@ def test_create_role_allow_list_fail(self):
         response = self.create_role(role_name, in_access_data=access_data)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_create_role_appfilter_fields_fail(self):
```
Comment (catastrophe-brandon) [inline tests/management/role/test_view.py:215]: In addition to these, is there a test for creation of a role with invalid keys in the attributeFilter object?
```
access_data = [
            {
                "permission": "cost-management:*:*",
                "resourceDefinitions": [{"attributeFilter": {"One": "keyA", "Two": "boop", "Three": "valueA"}}],
            }
        ]
```

Comment (catastrophe-brandon) [reply tests/management/role/test_view.py:215]: Ah, I see. I missed the "foo" key in the test. This should be fine for what we need.

File: tests/management/role/test_view.py [hunk 2]
```diff
@@ -212,6 +212,30 @@ def test_create_role_allow_list_fail(self):
...
+        """Test that we cannot create a role with an invalid operation."""
```
Comment (catastrophe-brandon) [inline tests/management/role/test_view.py:216]: Maybe `Test that we cannot create a role with an invalid key in the attributeFilter object` would be a better description? The operation value coded in the test looks valid.

**Review by astrozzc — APPROVED** [review]
Except the description of the test Brandon mentioned, LGTM!

**Review by coderbydesign — APPROVED** [review]
Needs a rebase, but looks good!

**Pull request merged**